### PR TITLE
【Hackathon 4 No.21】Add i1 / i1e to paddle

### PR DESCRIPTION
**Pull request by LyndonKong**
### PR types
New features

### PR changes
APIs

### Description
We add i1 and i1e ops for paddle.
document link is [here](https://github.com/PaddlePaddle/docs/pull/5839)
rfc link is [here](https://github.com/PaddlePaddle/community/pull/489)

### CODE REVIEW

**Comment by LyndonKong** [comment]
We are working on to fix the bugs in backward operators, not ready to be reviewed.

**Comment by LyndonKong** [comment]
Done with our local unittests, this PR is now ready to be reviewed @luotao1 .

**Comment by LyndonKong** [comment]
Backward bugs on GPU, try to test GPU in our local machine.

**Comment by zhengqiwen1997** [comment]
我们内部在审核i0/i0e的[PR](https://github.com/PaddlePaddle/Paddle/pull/52058)后，认为使用Eigen可能不支持后续高维度Tensor的开展，具体原因请见PR下方我的评论。所以需要麻烦你修改一下实现策略，可以考虑之前i0/i0e的实现方法，并且需要先修改rfc文档。为你的开发过程带来的不便深感抱歉！

**Comment by LyndonKong** [comment]
> 我们内部在审核i0/i0e的[PR](https://github.com/PaddlePaddle/Paddle/pull/52058)后，认为使用Eigen可能不支持后续高维度Tensor的开展，具体原因请见PR下方我的评论。所以需要麻烦你修改一下实现策略，可以考虑之前i0/i0e的实现方法，并且需要先修改rfc文档。为你的开发过程带来的不便深感抱歉！

OK, we will change our implementation to this version and update our RFC document in  a few days.

**Comment by LyndonKong** [comment]
This pr is ready to reviewed @zhengqiwen1997 . Our commit is rebased on the latest develop branch, but it seems that the CI is blocked due to other unrelated op tests. 
CI build result see [here](https://xly.bce.baidu.com/paddlepaddle/paddle/newipipe/detail/8357476/job/22636778)

image

**Comment by LyndonKong** [comment]
This pr has passed all CI checks, we would appreciate if you could review it @luotao1  @zhengqiwen1997 .

**Review by zhengqiwen1997 — APPROVED** [review]

**Review by zhengqiwen1997 — DISMISSED** [review]

**Comment by LyndonKong** [comment]
辛苦review @luotao1

**Comment by luotao1** [comment]
@LyndonKong 待 i0/i0e 技术 approve 后，会一起进入下一轮审核
- #52058

**Review by zhengqiwen1997 — DISMISSED** [review]

File: python/paddle/tensor/math.py [hunk 1]
```diff
@@ -5584,3 +5584,59 @@ def nextafter(x, y, name=None):
...
+def i1(x, name=None):
+    """
+    The function is used to calculate modified bessel function of order 1.
+    Args:
+        x (Tensor): The input tensor, it's data type should be float32, float64.
+        name (str, optional): For details, please refer to :ref:`api_guide_Name`. Generally, no setting is required. Default: None.
+    Returns:
+        - out (Tensor), A Tensor. the value of the modified bessel function of order 1 at x.
+    Examples:
+        .. code-block:: python
```
Comment (sunzhongkai588) [inline python/paddle/tensor/math.py:5674]: 在code-block下方空行，否则解析错误，参考paddle.add的写法
```
    Examples:
        .. code-block:: python

            import paddle

            x = paddle.to_tensor([2, 3, 4], 'float64')
            y = paddle.to_tensor([1, 5, 2], 'float64')
            z = paddle.add(x, y)
            print(z)  # [3., 8., 6. ]

```

File: python/paddle/tensor/math.py [hunk 2]
```diff
@@ -5584,3 +5584,59 @@ def nextafter(x, y, name=None):
...
+def i1e(x, name=None):
+    """
+    The function is used to calculate exponentially scaled modified Bessel function of order 1.
+    Args:
+        x (Tensor): The input tensor, it's data type should be float32, float64.
+        name (str, optional): For details, please refer to :ref:`api_guide_Name`. Generally, no setting is required. Default: None.
+    Returns:
+        - out (Tensor), A Tensor. the value of the exponentially scaled modified Bessel function of order 1 at x.
+    Examples:
+        .. code-block:: python
```
Comment (sunzhongkai588) [inline python/paddle/tensor/math.py:5708]: 同样的错误，需要加空行

File: python/paddle/tensor/math.py [hunk 3]
```diff
@@ -5584,3 +5584,59 @@ def nextafter(x, y, name=None):
         outputs = {"out": out}
         helper.append_op(type=op_type, inputs=inputs, outputs=outputs)
     return out
+
+
+def i1(x, name=None):
+    """
+    The function is used to calculate modified bessel function of order 1.
+    Args:
```
Comment (sunzhongkai588) [inline python/paddle/tensor/math.py:5666]: args上方加空行，否则解析错误

File: python/paddle/tensor/math.py [hunk 4]
```diff
@@ -5584,3 +5584,59 @@ def nextafter(x, y, name=None):
...
+        x (Tensor): The input tensor, it's data type should be float32, float64.
+        name (str, optional): For details, please refer to :ref:`api_guide_Name`. Generally, no setting is required. Default: None.
+    Returns:
```
Comment (sunzhongkai588) [inline python/paddle/tensor/math.py:5670]: returns上方加空行

File: python/paddle/tensor/math.py [hunk 5]
```diff
@@ -5584,3 +5584,59 @@ def nextafter(x, y, name=None):
...
+        - out (Tensor), A Tensor. the value of the modified bessel function of order 1 at x.
+    Examples:
```
Comment (sunzhongkai588) [inline python/paddle/tensor/math.py:5673]: examples上方加空行

File: python/paddle/tensor/math.py [hunk 6]
```diff
@@ -5584,3 +5584,59 @@ def nextafter(x, y, name=None):
...
+        helper.append_op(
+            type='i1', inputs={'x': x}, outputs={'out': out}, attrs={}
+        )
+    return out
+
+
+def i1e(x, name=None):
+    """
+    The function is used to calculate exponentially scaled modified Bessel function of order 1.
+    Args:
```
Comment (sunzhongkai588) [inline python/paddle/tensor/math.py:5699]: 上方加空行

File: python/paddle/tensor/math.py [hunk 7]
```diff
@@ -5584,3 +5584,59 @@ def nextafter(x, y, name=None):
...
+        x (Tensor): The input tensor, it's data type should be float32, float64.
+        name (str, optional): For details, please refer to :ref:`api_guide_Name`. Generally, no setting is required. Default: None.
+    Returns:
```
Comment (sunzhongkai588) [inline python/paddle/tensor/math.py:5704]: 上方加空行

File: python/paddle/tensor/math.py [hunk 8]
```diff
@@ -5584,3 +5584,59 @@ def nextafter(x, y, name=None):
...
+        - out (Tensor), A Tensor. the value of the exponentially scaled modified Bessel function of order 1 at x.
+    Examples:
```
Comment (sunzhongkai588) [inline python/paddle/tensor/math.py:5707]: 上方加空行

**Review by sunzhongkai588 — COMMENTED** [review]
格式请参考模板 https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/dev_guides/api_contributing_guides/api_docs_guidelines_cn.html#yingwenmuban

包括空行之类的格式需要严格对齐

**Comment by LyndonKong** [comment]
Fix the mentioned issue of doc-string in our new commit @sunzhongkai588

**Review by sunzhongkai588 — APPROVED** [review]
LGTM for docs

**Review by zhengqiwen1997 — APPROVED** [review]

**Review by jeff41404 — APPROVED** [review]
LGTM

**Review by zyfncg — APPROVED** [review]

**Pull request merged**